src/controllers/hid/hiddenylist.h

```diff
@@ -1,21 +1,37 @@
 #pragma once
 
-typedef struct hid_denylist {
-    unsigned short vendor_id;
-    unsigned short product_id;
-    unsigned short usage_page;
-    unsigned short usage;
-    int interface_number;
-} hid_denylist_t;
+#include <cstdint>
+
+// TODO: unify this with the invalid interfacenumber from the bulkenumerator
+constexpr static std::int32_t kInvalidInterfaceNumber = -1;
+constexpr static std::int32_t kBlanketValue = 0x0;
+
+struct hid_denylist_t {
+    std::uint16_t vendor_id;
+    std::uint16_t product_id;
+    std::uint16_t usage_page;
+    std::uint16_t usage;
+    std::int32_t interface_number = kInvalidInterfaceNumber;
```

Comment:
Why changing the type here? Isn't is safer to compare int with int than int with std::int32_t ?

Reply:
I assumed it was defined to always be a 32-bit int (similar to how the vid/pid are always 16-bit), though I didn't bother checking. Revert to int?

Reply:
I don't know any architecture we suport that defines `sizeof(int) != 32 bit` so it probably doesn't matter.

Reply:
Propably it doesn't matter, as common USB devices have only very few interfaces. But why change the type, if we just compare it? The header which defines it is here: https://github.com/libusb/hidapi/blob/c3c79a7d82933fb9642f89046745aacc5454f438/hidapi/hidapi.h#L178

Reply:
Well, my though process was basically that the data here carried is defined by a protocol that is [independent of the data model](https://en.cppreference.com/w/cpp/language/types#Properties). I assumed the data source had similar fixed-width definitions (I didn't look at the hidapi definition previously). I see it makes sense to match the "upstream" definition. I'll revert.